### PR TITLE
[ENG-7709][eas-cli][eas-json] add `m-medium` and `m-large` resource classes

### PR DESCRIPTION
**Pull request by szdziedzic**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

https://linear.app/expo/issue/ENG-7709/create-m-large-and-m-medium-resource-classes

# How

Add new `m-medium` and `m-large` resource classes.

Make `m1-medium` and `m1-experimental` point to `m-medium` on the server side.

Don't suggest `m1-medium` in the `eas.json` schema to users. Suggest `m-medium` instead.

I'm not adding `m-large` to `eas.schema.json`, because it's hidden behind the feature gate and it isn't publicly available.

# Test Plan

Run builds locally

### CODE REVIEW

File: packages/eas-cli/src/build/utils/resourceClass.ts [hunk 1]
```diff
@@ -8,24 +8,30 @@ import { BuildResourceClass } from '../../graphql/generated';
 import Log from '../../log';
 import { getReactNativeVersionAsync } from '../metadata';
 
+type AndroidResourceClass = Exclude<
```
Comment (dsokal) [inline packages/eas-cli/src/build/utils/resourceClass.ts:11]: Maybe `Pick<>` is better than `Exclude<>`? If we ever add an iOS resource class, you'll have to remember about updating this type.

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/build/utils/resourceClass.ts (hunk 1 above)
Comment (szdziedzic) [reply packages/eas-cli/src/build/utils/resourceClass.ts:11]: I think you can't do `Pick` on an enum in TS 🤔. Or maybe I'm wrong?

File: packages/eas-cli/src/build/utils/resourceClass.ts [hunk 2]
```diff
@@ -90,7 +88,19 @@ async function resolveIosResourceClassAsync(
...
+      )} instead.`
+    );
+  }
+
+  if (resourceClass === ResourceClass.M1_MEDIUM) {
+    Log.warn(
+      `Resource class ${chalk.bold('m1-medium')} is deprecated. Use ${chalk.bold(
+        'm-medium'
+      )} instead.`
+    );
```
Comment (szdziedzic) [inline packages/eas-cli/src/build/utils/resourceClass.ts:103]: I also added deprecation warning for `m1-medium`

File: packages/eas-cli/src/build/utils/resourceClass.ts [hunk 3]
```diff
@@ -8,24 +8,30 @@ import { BuildResourceClass } from '../../graphql/generated';
...
+>;
+
 const iosResourceClassToBuildResourceClassMapping: Record<ResourceClass, BuildResourceClass> = {
   [ResourceClass.DEFAULT]: BuildResourceClass.IosDefault,
   [ResourceClass.LARGE]: BuildResourceClass.IosLarge,
-  [ResourceClass.M1_EXPERIMENTAL]: BuildResourceClass.IosM1Large,
-  [ResourceClass.M1_MEDIUM]: BuildResourceClass.IosM1Medium,
+  [ResourceClass.M1_EXPERIMENTAL]: BuildResourceClass.IosMMedium,
+  [ResourceClass.M1_MEDIUM]: BuildResourceClass.IosMMedium,
   [ResourceClass.M1_LARGE]: BuildResourceClass.IosM1Large,
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/utils/resourceClass.ts:26]:
```suggestion
  [ResourceClass.M1_LARGE]: BuildResourceClass.IosMLarge,
```
Shouldn't this also be changed?

File: packages/eas-cli/src/build/configure.ts [hunk 4]
```diff
@@ -54,18 +54,18 @@ const EAS_JSON_MANAGED_DEFAULT: EasJson = {
       developmentClient: true,
       distribution: 'internal',
       ios: {
-        resourceClass: ResourceClass.M1_MEDIUM,
+        resourceClass: ResourceClass.M_MEDIUM,
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/configure.ts:57]: didn't we want to remove those defaults? Or do we want to remove them, but not just yet?

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/build/configure.ts (hunk 4 above)
Comment (szdziedzic) [reply packages/eas-cli/src/build/configure.ts:57]: They will be removed as a part of https://github.com/expo/eas-cli/pull/1734

**Comment by szdziedzic** [comment]
/changelog-entry new-feature Add new `m-medium` and `m-large` resource classes

**Pull request merged**